### PR TITLE
Replace custom mv logic for gram.c with make targets.

### PR DESCRIPTION
**Pull request by msharov**

### CODE REVIEW

**Comment by andmarti1424** [comment]
Thanks!

File: src/Makefile [hunk 1]
```diff
@@ -188,20 +188,23 @@ $(name)qref: sc.h
...
-	$(CC) ${CFLAGS} -c -DPSC pvmtbl.c
+gram.c : y.tab.c
+	mv $< $@
+
+y.tab.h : gram.y | gram.c
 
-experres.h : gram.y
-	sed -f eres.sed < gram.y > experres.h
+experres.h : gram.y eres.sed
+	sed -f eres.sed < $< > $@
```
Comment (mschilli87) [inline src/Makefile:201]:
```suggestion
	sed -f $(lastword $^) < $< > $@
```

File: src/Makefile [hunk 2]
```diff
@@ -188,20 +188,23 @@ $(name)qref: sc.h
...
 
-statres.h : gram.y
-	sed -f sres.sed < gram.y > statres.h
+statres.h : gram.y sres.sed
+	sed -f sres.sed < $< > $@
```
Comment (mschilli87) [inline src/Makefile:204]:
```suggestion
	sed -f $(lastword $^) < $< > $@
```

File: src/Makefile [hunk 3]
```diff
@@ -217,7 +220,7 @@ man_uninstall:
 
 clean:
 	rm -f $(OBJS)
-	rm -f *res.h y.tab.h
+	rm -f experres.h statres.h y.tab.h
```
Comment (mschilli87) [inline src/Makefile:223]: How is this related to changing the `mv` logic?

Anyhow, IMHO explicit is always better than implicit, especially when it comes to deleting things.
So I guess the change is fine but should maybe be mentioned in the commit message.

Comment (msharov) [reply src/Makefile:223]: I was uncomfortable with wildcards in rm -f; it may just end up deleting something you don't want deleted. I've been burned by this before. There is also a rm -rf ../docs at the end that I would have liked to eliminate. Explicitly listing what you want to delete would be much safer.

File: src/Makefile (hunk 1 above)
Comment (msharov) [reply src/Makefile:201]: That would not be more maintainable because other dependencies may end up added to the list, resulting in a very non-obvious problem:
```make
$(wildcard *.h): Makefile
```
